src/weather.cpp

```diff
@@ -652,6 +652,14 @@ std::string print_temperature( double fahrenheit, int decimals )
     }
 }
 
+std::string print_winddirection( int angle )
+{
+    std::ostringstream ret;
+    ret << std::fixed;
```

Comment:
`std::fixed` is unnecessary here as it's a floatfield format flag, no effect.

Also, we know `angle` should be within range `[0, 359]` (3 char) so it's probably faster to just use string here. `std::ostringstream` does a lot of work for such a simple operation.
```
// Using std::string only
std::string print_winddirection( int angle )
{
    assert( angle >= 0 && angle < 360 && "Angle out of bounds [0, 359]" ); // Assert on bad input
    // Ensure size, angle is in range [0, 359], 3 characters + null terminator
    std::string ret( 4, '\0' ); // Might get small string optimization and not heap allocate
    snprintf( &ret.front(), 4, "%d", angle );
    return string_format( pgettext( "wind direction angle", "%s" ), ret.c_str() );
}
```
Or using a char buffer
```
// Using char buffer
std::string print_winddirection( int angle )
{
    assert( angle >= 0 && angle < 360 && "Angle out of bounds [0, 359]" ); // Assert on bad input
    char buf[4] = ""; // Stack allocation of buffer
    snprintf( buff, 4, "%d", angle );
    return string_format( pgettext( "wind direction angle", "%s" ), buf );
}
```
Both alternatives do significantly less work than using `std::ostringstream`.

Reply:
I think that entire function is not necessary now, it's a leftover from when I was trying different ways to print the direction string.  The CBM and weather reader iuse just do ```add_msg_if_player( m_info, _( "Wind Direction: From the %s." ), weatherPoint.dirstring );``` 
I can just remove that function? or can you think of a reason to keep it there?

Reply:
No idea if it's useful to keep around, just noticed that it wasn't actually used anywhere haha. Using dirstring is probably better, unless the player has a way to tell the exact direction an 8-way discrete representation works